### PR TITLE
Improved: update page

### PR DESCRIPTION
**Pull request by math-GH**
Goal: The update page has the default style and looks more clean.

The update mechanism itself was not touched

Before:
grafik

grafik

After:
- a link to the changelog added
- show the release channel (`Development/Nightly` or `Stable`), based on the current version number (Development = with `-dev`, stable = without `-dev`)
- blue button
- button in the current style for submit buttons
- default "table" layout
- renamed a bit the buttons
grafik

The branch name is shown, if it will update via git
grafik

When user clicks on the `Start update` button, then the button turns into a grey deactivated button inkl. a loading gif until the update is done.
grafik

Here a picture if the update is via update.freshrss.org and not via git:
grafik

How it looks like if it is a `stable` version
grafik

grafik

While/After updating an entry will added to the log with information about the update.

How to test the feature manually:

How to manipulate:
return false = it will use the update via update.freshrss.org
https://github.com/FreshRSS/FreshRSS/blob/445cc23abdda5767b622d70cf7b5eb5310dcf908/app/Controllers/updateController.php#LL7C1-L9C3

release channel (`Development/Nightly` or `Stable`), based on the current version number (Development = with `-dev`, stable = without `-dev`)
https://github.com/FreshRSS/FreshRSS/blob/445cc23abdda5767b622d70cf7b5eb5310dcf908/constants.php#L6

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: app/Controllers/updateController.php [hunk 1]
```diff
@@ -44,6 +44,16 @@ public static function migrateToGitEdge(): bool {
 		return true;
 	}
 
+	public static function getCurrentGitBranch(): string {
+		$output = array();
```
Comment (ColonelMoutarde) [inline app/Controllers/updateController.php]:
```suggestion
		$output = [];
```

File: app/Controllers/updateController.php [hunk 2]
```diff
@@ -44,6 +44,16 @@ public static function migrateToGitEdge(): bool {
...
+		exec('git branch --show-current', $output, $return);
+		if ($return == 0) {
```
Comment (ColonelMoutarde) [inline app/Controllers/updateController.php]:
```suggestion
		if ($return === 0) {
```

File: app/Controllers/updateController.php [hunk 3]
```diff
@@ -144,7 +157,20 @@ public function indexAction(): void {
 		}
 	}
 
+	private function is_release_channel_stable(string $currentVersion): bool {
+		if (strpos($currentVersion, 'dev') > 0) {
+			return false;
+		} else {
+			return true;
+		}
```
Comment (ColonelMoutarde) [inline app/Controllers/updateController.php]:
```suggestion
		return (strpos($currentVersion, 'dev') > 0) ? false : true;
		}
```

**Review by ColonelMoutarde — COMMENTED** [review]
Small recommendations ;)

**Comment by Alkarex** [comment]
I have updated a bit the terminology https://github.com/FreshRSS/FreshRSS/pull/5420/commits/8dd4a38ea7105996b428251998583f5a1c000d76

**Comment by math-GH** [comment]
Ready for final review

**Pull request merged**